beacon_node/network/src/beacon_processor/work_reprocessing_queue.rs

```diff
@@ -65,6 +69,9 @@ const MAXIMUM_QUEUED_ATTESTATIONS: usize = 16_384;
 /// How many light client updates we keep before new ones get dropped.
 const MAXIMUM_QUEUED_LIGHT_CLIENT_UPDATES: usize = 128;
 
+// Process backfill batch 50%, 60%, 80% through each slot
+pub const BACKFILL_SCHEDULE_IN_SLOT: [f32; 3] = [0.5f32, 0.6, 0.8];
```

Comment:
When doing "fractions of a slot" logic I've traditionally avoided using floats since they a bit more challenging to reason about than ints due to float's inherent complexity and implementation quirks between languages\platforms. Here's a suggestion on how we could achieve the same thing with only ints:


```suggestion
// Process backfill batch 50%, 60%, 80% through each slot.
//
// Note: use caution to set these fractions in a way that won't cause panic-y
// arithmetic.
pub const BACKFILL_SCHEDULE_IN_SLOT: [(u64, u64); 3] = [
    // One half: 6s on mainnet, 2.5s on Gnosis.
    (1, 2),
    // Three fifths: 7.2s on mainnet, 3s on Gnosis.
    (3, 5),
    // Four fifths: 9.6s on mainnet, 4s on Gnosis.
    (4, 5),
];
```

Reply:
Yep makes a lot of sense, thanks for adding the comments!

Reply:
I'm updating your suggested type to `[(u32, u32); 3]` so it works with duration without casting, I think u32 would be enough for us unless I'm missing something?

Reply:
> I'm updating your suggested type to `[(u32, u32); 3]` so it works with duration without casting, I think u32 would be enough for us unless I'm missing something?

Oh yep nice!